src/QMCWaveFunctions/Fermion/MultiDiracDeterminant.2.cpp

```diff
@@ -189,14 +189,14 @@ void MultiDiracDeterminant::BuildDotProductsAndCalculateRatios(int ref,
 #endif
 }
 
-void MultiDiracDeterminant::evaluateDetsForPtclMove(ParticleSet& P, int iat)
+void MultiDiracDeterminant::evaluateDetsForPtclMove(const ParticleSet& P, int iat, int refPtcl)
 {
   UpdateMode = ORB_PBYP_RATIO;
   RatioTimer.start();
   evalOrbTimer.start();
   Phi->evaluateValue(P, iat, psiV);
   evalOrbTimer.stop();
-  WorkingIndex = iat - FirstIndex;
+  WorkingIndex = (refPtcl < 0 ? iat : refPtcl) - FirstIndex;
```

Comment:
So what insures  
```
that
FirstIndex  is never > iat or refPtcl
```

I've seen this cause mayhem elsewhere.

Reply:
Good to have. Will add.

Reply:
Added assert for all the similar places.